python/ray/experimental/sgd/pytorch/examples/cifar_pytorch_example.py

```diff
@@ -96,23 +48,33 @@ def optimizer_creator(model, config):
     return torch.optim.SGD(model.parameters(), lr=config.get("lr", 0.1))
 
 
-def train_example(num_replicas=1, use_gpu=False, test_mode=False):
-    config = {"test_mode": test_mode}
+def scheduler_creator(optimizer, config):
+    return torch.optim.lr_scheduler.MultiStepLR(
+        optimizer, milestones=[150, 250, 350], gamma=0.1)
+
+
+def train_example(num_replicas=1,
+                  num_epochs=5,
+                  use_gpu=False,
+                  use_fp16=False,
+                  test_mode=False):
+    config = {TEST_MODE: test_mode}
     trainer1 = PyTorchTrainer(
         ResNet18,
         cifar_creator,
         optimizer_creator,
         nn.CrossEntropyLoss,
+        scheduler_creator=scheduler_creator,
         initialization_hook=initialization_hook,
-        train_function=train,
-        validation_function=validate,
         num_replicas=num_replicas,
         config=config,
         use_gpu=use_gpu,
         batch_size=16 if test_mode else 512,
-        backend="nccl" if use_gpu else "gloo")
-    for i in range(5):
-        stats = trainer1.train()
+        backend="nccl" if use_gpu else "gloo",
+        scheduler_step_freq="epoch",
+        use_fp16=use_fp16)
+    for i in range(num_epochs):
+        stats = trainer1.train(max_retries=0)
```

Comment:
Is there a reason for explicitly specifying the default option here?

Reply:
Thanks - made it more explicit to point out to users when to change `max_retries`.